### ISSUE TITLE
"Access denied" error disappears after executing command for the 2nd time

### ISSUE THREAD

**Issue by kokosek** [drush-ops/drush#2218]
I'm using drush to reinstall a module so that .install is executed.
I run:
`drush pmu medialib
drush en medialib`
...and I get an error on after the latter command:

> The following extensions will be enabled: medialib
> Do you really want to continue? (y/n): y
> WD node: Exception: **Access denied**. in medialib_node_presave() (line 381 of                                    [error]
> /srv/www/htdocs/project/sites/all/modules/medialib/medialib.module).
> exception 'Exception' with message 'Access denied.' in                                                            [error]
> /srv/www/htdocs/project/sites/all/modules/medialib/medialib.module:381
> Stack trace:
> #0 [internal function]: medialib_node_presave(Object(stdClass))
> #1 /srv/www/htdocs/project/includes/module.inc(951): call_user_func_array('medialib_no...', Array)
> #2 /srv/www/htdocs/project/modules/node/node.module(1106): module_invoke_all('node_presave', Object(stdClass))
> #3 /srv/www/htdocs/project/sites/all/modules/medialib/medialib.api_common.inc(538):
> node_save(Object(stdClass))
> #4 /srv/www/htdocs/project/sites/all/modules/medialib/medialib.install(390):
> _medialib_create_directory_raw('Media Library', NULL)
> #5 [internal function]: medialib_enable()
> #6 /srv/www/htdocs/project/includes/module.inc(922): call_user_func_array('medialib_en...', Array)
> #7 /srv/www/htdocs/project/includes/module.inc(497): module_invoke('medialib', 'enable')
> #8 phar:///usr/local/bin/drush/commands/core/drupal/environment_7.inc(143): module_enable(Array)
> #9 phar:///usr/local/bin/drush/commands/pm/pm.drush.inc(1167): drush_module_enable(Array)
> #10 [internal function]: drush_pm_enable('medialib')
> #11 phar:///usr/local/bin/drush/includes/command.inc(366): call_user_func_array('drush_pm_enable', Array)
> #12 phar:///usr/local/bin/drush/includes/command.inc(217): _drush_invoke_hooks(Array, Array)
> #13 [internal function]: drush_command('medialib')
> #14 phar:///usr/local/bin/drush/includes/command.inc(185): call_user_func_array('drush_command', Array)
> #15 phar:///usr/local/bin/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
> #16 phar:///usr/local/bin/drush/includes/preflight.inc(66): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
> #17 phar:///usr/local/bin/drush/includes/startup.inc(325): drush_main()
> #18 phar:///usr/local/bin/drush/drush(114): drush_startup(Array)
> #19 /usr/local/bin/drush(10): require('phar:///usr/loc...')
> #20 {main}

...but when I execute the same command again I get:

> array_values() expects parameter 1 to be array, null given library.inc.php:123                                    [warning]
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The Betterawstats engine generated the error below. If it is a configuration error concerning AWStats locations   [error]
> (i.e. the AWstats data directory, library directory, or language direction), these must be set using the
> Statystyki BAW administration pages.
> The language file could not be found in this location: /usr/share/awstats/lang/awstats-en.txtPlease check your    [warning]
> config.php. The value of $BAW_CONF['path_lang'] has to point to the directory where awstats stores its language
> files (typically somewhere /cgi-bin/awstats/lang). Please also make sure that php has read-permissions on that
> folder.
> The library files could not be found in this location: /usr/share/awstats/lib/operating_systems.pmPlease check    [warning]
> your config.php. The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library
> files (typically somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that
> folder.
> The library files could not be found in this location: /usr/share/awstats/lib/domains.pmPlease check your         [warning]
> config.php. The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library
> files (typically somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that
> folder.
> The library files could not be found in this location: /usr/share/awstats/lib/robots.pmPlease check your          [warning]
> config.php. The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library
> files (typically somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that
> folder.
> The library files could not be found in this location: /usr/share/awstats/lib/worms.pmPlease check your           [warning]
> config.php. The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library
> files (typically somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that
> folder.
> The library files could not be found in this location: /usr/share/awstats/lib/mime.pmPlease check your config.php.[warning]
> The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library files (typically
> somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that folder.
> The library files could not be found in this location: /usr/share/awstats/lib/mime.pmPlease check your config.php.[warning]
> The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library files (typically
> somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that folder.
> The library files could not be found in this location: /usr/share/awstats/lib/browsers.pmPlease check your        [warning]
> config.php. The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library
> files (typically somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that
> folder.
> The library files could not be found in this location: /usr/share/awstats/lib/search_engines.pmPlease check your  [warning]
> config.php. The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library
> files (typically somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that
> folder.
> The library files could not be found in this location: /usr/share/awstats/lib/status_http.pmPlease check your     [warning]
> config.php. The value of $BAW_CONF['path_lib'] has to point to the directory where awstats stores its library
> files (typically somewhere /cgi-bin/awstats/lib). Please also make sure that php has read-permissions on that
> folder.

These AWSTATS errors doesn't matter. I get them all the time because I don't have AWSTATS installed. I didn't remove them just in case.

**To sum up:** executing drush command for multiple times changes the outcome which is not a good practise.
